### PR TITLE
feature/idea ultimate aarch64 linux

### PR DESCRIPTION
**Pull request by sequencer**
- jetbrains.idea-ultimate: add aarch64-linux as a supported platform
- Removed unused input
- update and fix aarch64 intellij versions

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/jetbrains-jdk/default.nix [hunk 1]
```diff
@@ -57,22 +57,22 @@ openjdk17.overrideAttrs (oldAttrs: rec {
   buildPhase = ''
     runHook preBuild
 
-    mkdir -p jcef_linux_x64/jmods
-    cp ${jetbrains.jcef}/* jcef_linux_x64/jmods
+    mkdir -p jcef_linux_${if stdenv.isAarch64 then "aarch64" else "x64"}/jmods
```
Comment (NickCao) [inline pkgs/development/compilers/jetbrains-jdk/default.nix:60]: Can we de-duplicate these `if stdenv.isAarch64 then "aarch64" else "x64"`

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 2]
```diff
@@ -73,6 +72,25 @@ let rpath = lib.makeLibraryPath [
 ];
 
 buildType = if debugBuild then "Debug" else "Release";
+  lookupTable = {
```
Comment (NickCao) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:75]: The indentation is a bit off.

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 3]
```diff
@@ -122,13 +139,13 @@ in stdenv.mkDerivation rec {
...
 
-    cp ${clang-fmt} tools/buildtools/linux64/clang-format
+    cp ${clang-format} tools/buildtools/linux64/clang-format
     chmod +w tools/buildtools/linux64/clang-format
 
     mkdir jcef_build
     cd jcef_build
 
-    cmake -G "Ninja" -DPROJECT_ARCH="x86_64" -DCMAKE_BUILD_TYPE=${buildType} ..
+    cmake -G "Ninja" -DPROJECT_ARCH="${if stdenv.isAarch64 then "arm64" else "x86_64"}" -DCMAKE_BUILD_TYPE=${buildType} ..
```
Comment (NickCao) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:148]: Also dedup these.

File: pkgs/development/compilers/jetbrains-jdk/jcef.nix [hunk 4]
```diff
@@ -183,7 +200,12 @@ in stdenv.mkDerivation rec {
     jar uf gluegen-rt.jar module-info.class
     rm module-info.class module-info.java
     mkdir lib
-    extract_jar "$JOGAMP_DIR"/gluegen-rt-natives-"$OS"-"$DEPS_ARCH".jar lib natives/"$OS"-"$DEPS_ARCH"
+
+
+    if [ "$OS" == "macosx" ] || [ "$TARGET_ARCH" == "x86_64" ]; then
```
Comment (NickCao) [inline pkgs/development/compilers/jetbrains-jdk/jcef.nix:205]: Add comment about why only these platforms need special handling.

**Comment by GenericNerdyUsername** [comment]
rebase?

**Comment by sequencer** [comment]
> rebase?

Rebased

**Comment by wegank** [comment]
Merge conflict, please rebase.

**Pull request closed without merging**